src/Control/Monad/Logic/Sequence/Internal.hs

```diff
@@ -181,7 +181,7 @@ toView (SeqT s) = case S.viewl s of
   S.EmptyL -> return Empty
   h S.:< t -> h >>= \x -> case x of
     Empty -> toView (SeqT t)
-    hi :< SeqT ti -> return (hi :< SeqT (ti S.>< t))
+    hi :< SeqT ti -> return (hi :< SeqT (ti S.|> toView (SeqT t)))
```

Comment:
This makes me nervous. Might well be the wrong asymptotics.

Reply:
Actually... Probably fine. Will need to analyze further.

Reply:
No, forget it...

Reply:
I'm late to the party. What did you learn? Bad asymptotics?

Reply:
Yeah, I'm pretty sure it's wrong. I do want to go back to the idea of unpacking the simple queue into the catenable one. And maybe try adding some special constructors for certain things; not sure.